packages/ui/src/hooks/use-social.ts

```diff
@@ -109,6 +109,12 @@ const useSocial = (options?: Options) => {
     signInWithSocialErrorHandlers
   );
 
+  /* 
+    This is needed because the callback useEffect handlers can not get the request parameters
+    Hacky solution. Need to be refactored. This is OK only because our requests are running synchronously
+  */
+  const invokedConnectorIdRef = useRef<string>();
```

Comment:
I think the useEffect code can be moved into the useCallback below, then we can avoid using this ref. Thoughts?

Reply:
fixed